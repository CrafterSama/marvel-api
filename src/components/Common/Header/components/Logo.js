import React from 'react'
import { Link, Redirect } from 'react-router-dom'
import logo from '../../../../assets/marvel-logo.svg'
import { LogoContainer, ImageContainer } from '../../../../styles'
import { useStateValue } from '../../../../context'
import { getCharactersRandomly } from '../../../../context/actions'

const Logo = () => {

  const [, dispatch] = useStateValue()

  const toHome = () => {
    getCharactersRandomly(dispatch);
    return <Redirect to='/' />;
  }

  return (
    <LogoContainer>
      <ImageContainer>
        <Link to='/' onClick={toHome}>
          <img src={logo} alt='Marvel Logo' />
        </Link>
      </ImageContainer>
    </LogoContainer>
  );
}

export default Logo;
