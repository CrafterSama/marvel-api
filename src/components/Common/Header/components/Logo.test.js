import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import React from 'react';

import Logo from './Logo';

test('matches snapshot', () => {
  const wrapper = shallow(<Logo />);

  expect(toJSON(wrapper)).toMatchSnapshot();
})
