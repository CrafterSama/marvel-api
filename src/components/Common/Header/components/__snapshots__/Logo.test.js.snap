// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<styles__LogoContainer>
  <styles__ImageContainer
    darkmode=""
  >
    <Link
      onClick={[Function]}
      to="/"
    >
      <img
        alt="Marvel Logo"
        src={
          Object {
            "getCacheKey": [Function],
            "process": [Function],
          }
        }
      />
    </Link>
  </styles__ImageContainer>
</styles__LogoContainer>
`;
